addons/calendar/models/calendar_event.py

```diff
@@ -796,6 +796,29 @@ def _compute_field_value(self, field):
             return super(Meeting, self.with_context(prefetch_fields=False))._compute_field_value(field)
         return super()._compute_field_value(field)
 
+    @api.model
+    def search_read(self, domain=None, fields=None, offset=0, limit=None, order=None):
+        """
+        Complete the domain to respect privacy
+            - public + busy --> ok
+            - public + free --> ok
+            - private + busy --> ok (will be obfuscated)
+            - private + free --> ok if we are the user of the event
+        """
+        privacy_domain = [
+            '|',
+                ['privacy', '=', 'public'],
+                '&',
+                    ['privacy', '=', 'private'],
+                    '|',
+                        ['show_as', '=', 'busy'],
+                        '&',
+                            ['show_as', '=', 'free'],
+                            ['user_id', '=', self.env.user.id],
+        ]
+        domain = domain + privacy_domain
```

Comment:
you can use domain = expression.AND([...])
There is no test for this fix. We can't merge such change in stable without a test.